api.go

```diff
@@ -19,13 +19,14 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
-	"github.com/pivotal-cf/brokerapi/middlewares/originating_identity_header"
 	"net/http"
 	"strconv"
 
 	"code.cloudfoundry.org/lager"
 	"github.com/gorilla/mux"
-	"github.com/pivotal-cf/brokerapi/auth"
+	"github.com/portworx/brokerapi/auth"
```

Comment:
You shouldn't change these. They should still be `pivotal-cf`

Reply:
I think we should refer the auth, middleware and other local packages from the forked out repo, rather than adding the originating repo as a dependency thus causing confusion as to which package is being referred in the forked code.

Reply:
There are fixes in the local code, i.e structs and middlewares have been added to or re-ordered. I would prefer that the fork'd out repo refer's to these local packages rather than 3rd-party or the base repo, as this create an unnecessary hop and dependency to vendor it in, and debugging will be a huge issue as we would not know which package was referred to.
The way go code works, it would require the base repo to be either cloned or vendor'd in which is what I do not want, I want this fork to be fairly independent and when we want to forward merge these fixes, that can be taken care of in the main branch.

Reply:
Articles pointing out how to manage forks in go. The third link specifically points out what you want to do as a bad solution.
https://blog.gopheracademy.com/advent-2017/managing-dependencies-forks-and-code-patches-with-dep/
https://medium.com/@sasom/how-to-fork-go-packages-f863e370bbb6
http://code.openark.org/blog/development/forking-golang-repositories-on-github-and-managing-the-import-path

Reply:
I have seen these links, none of them provide an elegant solution and the solutions they provide convolute the development environment even more for debugging or other purposes. The last link just says that,
Hey! it is a dirty solution and goes on to give even pathetic solutions to the problem. Like clone your repo and rename the path or give half baked solutions of adding a remote to update the code from the base repo, but does not solve the problem of 50% packages referring to `github.com/awesome-you/proj` and rest referring to `github.com/awesome-org/proj`.
The only elegant and good solution to such an issue is to clone/fork the repo make changes and maintain them using `glide` as the dependency manager where we can do the following
```
package: github.com/awesome-org/proj
import:
- package: github.com/awesome-org/proj
  repo:    https://github.com/awesome-you/proj.git    <<--- note the repo is different
```
what this does is what we have done here but in an literal sense, we have cloned the repo and made to it refer our packages rather than the base repo's. Even for doing something like the above we will need to change the local package reference.
If we see the changes done here, they are all related to the packages local to `github.com/portworx/brokerapi`. So, the structs that we change under this repo should be referred to all the local packages and vice versa.